emissionsapi/preprocess.py

```diff
@@ -120,21 +119,8 @@ def filter_data(data, qa_percent):
     :return: scan object with filtered data
     :rtype: emissionsapi.preprocess.Scan
     """
-
-    # cast Scan objects to float because numpy.nan is float
-    data.quality = data.quality.astype('float')
-    data.data = data.data.astype('float')
-    data.longitude = data.longitude.astype('float')
-    data.latitude = data.latitude.astype('float')
```

Comment:
we don't need to cast to float anymore? i thought numpy.nan is of type float

Reply:
We never did for most since they are already float values. If you want to test this, try:

```python
print(type(data.data[0][0]))
# prints <class 'numpy.ndarray'>
```

For the rest, the reasoning is that we don't need to set all the values since we can identify the invalid measurements by looking at the measurements alone. Why check the coordinates as well?

Reply:
thats true